Sources/PackageCollections/Providers/PackageMetadataProvider.swift

```diff
@@ -16,23 +16,18 @@ import TSCBasic
 import TSCUtility
 
 /// `PackageBasicMetadata` provider
-protocol PackageMetadataProvider: Closable {
-    /// The name of the provider
-    var name: String { get }
-
-    /// Retrieves metadata for a package at the given repository address.
+protocol PackageMetadataProvider {
+    /// Retrieves metadata for a package with the given identity and repository address.
     ///
     /// - Parameters:
     ///   - identity: The package's identity
     ///   - location: The package's location
     ///   - callback: The closure to invoke when result becomes available
-    func get(identity: PackageIdentity, location: String, callback: @escaping (Result<PackageCollectionsModel.PackageBasicMetadata, Error>) -> Void)
-
-    /// Returns `AuthTokenType` for a package.
-    ///
-    /// - Parameters:
-    ///   - location: The package's location
-    func getAuthTokenType(for location: String) -> AuthTokenType?
+    func get(
+        identity: PackageIdentity,
+        location: String,
+        callback: @escaping (Result<PackageCollectionsModel.PackageBasicMetadata, Error>, PackageMetadataProviderContext?) -> Void
```

Comment:
In additional to the new APIs, I am making major changes around `PackageMetadataProviderContext` as well. 

The sole(?) purpose of `PackageMetadataProviderContext` is a way of telling the caller any auth-related warning (e.g., it's better to configure auth token for a higher API limit) or error (e.g., permission denied). i.e., `PackageMetadataProviderContext` is not just for error cases only. `PackageMetadataProviderContext` was constructed separately in previous logic, now it's part of the `get` response.

Also, `PackageMetadataProviderContext` has a new `name` property so the caller would know where the metadata comes from.